tensorrt_llm/llmapi/mpi_session.py

```diff
@@ -263,28 +264,54 @@ class RemoteMpiCommSessionClient():
     RemoteMpiCommSessionClient is a variant of MpiCommSession that is used to connect to a remote MPI pool.
     '''
 
-    def __init__(self, addr: str, hmac_key: Optional[bytes] = None):
+    def __init__(
+        self,
+        *,
+        addr: Optional[tuple[str, Optional[bytes]]] = None,
+        addr_file: Optional[str] = None,
+    ):
+        '''
+        Args:
+            addr: The address of the remote MPI pool.
+            addr_file: The file to write the address of the remote MPI pool to share with other intra-node processes.
+        '''
         # FIXME: this is a hack to avoid circular import, resolve later
         from tensorrt_llm.executor.ipc import ZeroMqQueue
-        self.addr = addr
+        self._is_shutdown = False
         print_colored_debug(
             f"RemoteMpiCommSessionClient connecting to {addr}\n", "yellow")
-        self.queue = ZeroMqQueue((addr, hmac_key),
-                                 is_server=False,
-                                 use_hmac_encryption=bool(hmac_key))
-        self._is_shutdown = False
+        self.queue = ZeroMqQueue(addr, is_server=True, use_hmac_encryption=True)
+        self.addr_file = addr_file
+        self.write_address_to_shared_file(
+        )  # for client process to get the address
 
     def submit(self, task: Callable[..., T], *args, **kwargs) -> list:
         if self._is_shutdown:
             print_colored_debug(
                 "RemoteMpiCommSessionClient is already shut down\n", "yellow")
             return []
         print_colored_debug(
-            f"RemoteMpiCommSessionClient [rank{global_mpi_rank()}] sending task {task} to {self.addr}\n",
+            f"RemoteMpiCommSessionClient [rank{global_mpi_rank()}] sending task {task} to {self.queue.address}\n",
             "yellow")
         self.queue.put(RemoteTask(task, args, kwargs))
         return []
 
+    def write_address_to_shared_file(self):
+        # This file will be deleted when client is destroyed, that is sufficient for client to read it during setup
+        with open(self.addr_file, "w") as f:
+            ip_addr, hmac_key = self.queue.address
+            if hmac_key is not None:
+                hmac_key = base64.b64encode(hmac_key).decode('ascii')
+            json.dump((ip_addr, hmac_key), f)
+
+    @staticmethod
+    def read_address_from_shared_file(path: str):
```

Comment:
Maybe delete it here after the file is used, such that you dont need to `rm` in a shell.